src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,53 @@
-import React from 'react';
+/* eslint-disable react/button-has-type */
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { CommentData } from '../types/Comment';
+
+interface NewCommentFormProps {
+  commentAdd: (comment: CommentData) => Promise<void>,
+}
+
+export const NewCommentForm: React.FC<NewCommentFormProps> = ({
+  commentAdd,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [commentLoading, setIsCommentLoading] = useState(false);
+  const [validSubmit, setValidSubmit] = useState(true);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (name && email && body) {
+      const newComment = {
+        name,
+        email,
+        body,
+      };
+
+      setIsCommentLoading(true);
+      setValidSubmit(true);
+
+      commentAdd(newComment)
+        .then(() => setBody(' '))
```

Comment:
In the .then() callback, you should setBody('') to clear the input instead of setting it to a whitespace.